tests/game/type-advantage.test.ts

```diff
@@ -0,0 +1,72 @@
+import {describe, expect, test} from '@jest/globals'
+import HotguyCommon from 'common/cards/alter-egos-iii/hermits/hotguy-common'
+import PoultrymanCommon from 'common/cards/alter-egos/hermits/poultryman-common'
+import {Hermit} from 'common/cards/base/types'
+import GoodTimesWithScarCommon from 'common/cards/default/hermits/goodtimeswithscar-common'
+import GrianCommon from 'common/cards/default/hermits/grian-common'
+import IJevinCommon from 'common/cards/default/hermits/ijevin-common'
+import ImpulseSVCommon from 'common/cards/default/hermits/impulsesv-common'
+import MumboJumboCommon from 'common/cards/default/hermits/mumbojumbo-common'
+import TinFoilChefCommon from 'common/cards/default/hermits/tinfoilchef-common'
+import XBCraftedCommon from 'common/cards/default/hermits/xbcrafted-common'
+import {RowComponent} from 'common/components'
+import query from 'common/components/query'
+import {attack, endTurn, playCardFromHand, testGame} from './utils'
+
+var typeToHermitMap: Map<String, Hermit> = new Map([
+	['Prankster', PoultrymanCommon],
+	['PvP', XBCraftedCommon],
+	['Redstoner', MumboJumboCommon],
+	['Speedrunner', HotguyCommon],
+	['Farm', ImpulseSVCommon],
+	['Explorer', IJevinCommon],
+	['Builder', GrianCommon],
+	['Terraformer', GoodTimesWithScarCommon],
+	['Miner', TinFoilChefCommon],
+])
+
+function createTypeAdvantageTest(attackingType: String, defendingType: String) {
+	test(attackingType + ' v ' + defendingType, () => {
+		let attacker = typeToHermitMap.get(attackingType)!
+		let defender = typeToHermitMap.get(defendingType)!
```

Comment:
This should work with Lunar's suggestion
```suggestion
type HermitType = keyof typeof typeToHermitMap

function createTypeAdvantageTest(attackingType: HermitType, defendingType: HermitType) {
	test(attackingType + ' v ' + defendingType, () => {
		const attacker = typeToHermitMap[attackingType]
		const defender = typeToHermitMap[defendingType]
```